app/concepts/reservations/use_cases/update_reservation.rb

```diff
@@ -0,0 +1,34 @@
+module Reservations
+  module UseCases
+    class UpdateReservation
+      def initialize(reservation, status)
+        @reservation = reservation
+        @status = status
+      end
+
+      def call
+        confirm_reservation if status == 'confirmed'
+        cancel_reservation if status == 'cancelled'
+      end
+
+      private
+
+      def confirm_reservation
+        reservation.update(status: Reservation::CONFIRMED)
+      end
+
+      def cancel_reservation
+        ActiveRecord::Base.transaction do
+          reservation.update!(status: Reservation::CANCELLED)
+          tickets.each(&:destroy!)
```

Comment:
👏🏽 
I like this idea of changing only status without deleting reservations from database but also making seats available again by destroying tickets